client/cypress/integration/query/parameter_spec.js

```diff
@@ -225,11 +283,114 @@ describe('Parameter', () => {
         .click()
         .click(); // workaround for datepicker display bug
 
+      cy.getByTestId('ParameterApplyButton')
+        .click();
+
       cy.get('@Input').then(($input) => {
         const now = Cypress.moment($input.val(), 'DD/MM/YY HH:mm');
         cy.getByTestId('DynamicTable')
           .should('contain', now.format('YYYY-MM-01 HH:mm'));
       });
     });
+
+    it('sets dirty state when edited', () => {
+      expectDirtyStateChange(() => {
+        cy.getByTestId('ParameterName-test-parameter')
+          .find('input')
+          .click();
+
+        cy.get('.ant-calendar-date-panel')
+          .contains('Now')
+          .click();
+      });
+    });
+  });
+
+  describe('Apply Changes', () => {
+    const expectAppliedChanges = (apply) => {
+      cy.getByTestId('ParameterName-test-parameter-1')
+        .find('input')
+        .as('Input')
+        .type('Redash');
+
+      cy.getByTestId('ParameterName-test-parameter-2')
+        .find('input')
+        .type('Redash');
+
+      cy.location('search').should('not.contain', 'Redash');
+
+      cy.server();
+      cy.route('POST', 'api/queries/*/results').as('Results');
+
+      apply(cy.get('@Input'));
+
+      cy.location('search').should('contain', 'Redash');
+      cy.wait('@Results');
+    };
+
+    beforeEach(() => {
+      const queryData = {
+        name: 'Testing Apply Button',
+        query: "SELECT '{{test-parameter-1}} {{ test-parameter-2 }}'",
```

Comment:
It seems you also need to provide the `options.parameters`. I've seen something related, not sure where :thinking:

Reply:
Yeah, I think @rauchy had a similar solution lately. Thanks!